lib/vanagon/platform/windows.rb

```diff
@@ -34,6 +34,143 @@ def package_name(project)
+  $toolsFolder = Join-Path $tempDir "tools"
+  $chocInstallPS1 = Join-Path $toolsFolder "chocolateyInstall.ps1"
+
+  if ($PSVersionTable.psversion.Major -gt 2) {
+    & $chocInstallPS1
+  }
+  else {
+    $output = Invoke-Expression $chocInstallPS1
+    $output
+    write-output "Any errors that occured during install or upgrade are logged here: $chocoErrorLog"
+    $error | out-file $chocErrorLog
+  }
+
+  $chocInstallVariableName = "ChocolateyInstall"
+  $chocoPath = [Environment]::GetEnvironmentVariable($chocInstallVariableName, [System.EnvironmentVariableTarget]::User)
+  if ($chocoPath -eq $null -or $chocoPath -eq '') {
+    $chocoPath = 'C:\\ProgramData\\Chocolatey'
+  }
+
+  Write-Output 'Ensuring chocolatey.nupkg is in the lib folder'
+  $chocoPkgDir = Join-Path $chocoPath 'lib\\chocolatey'
+  $nupkg = Join-Path $chocoPkgDir 'chocolatey.nupkg'
+  if (![System.IO.Directory]::Exists($chocoPkgDir)) { [System.IO.Directory]::CreateDirectory($chocoPkgDir); }
+  Copy-Item "$file" "$nupkg" -Force -ErrorAction SilentlyContinue
+}
+catch
+{
+  Write-Host "$($_.Exception.Message)"
+  exit 1
+}
+        BATCH
+      end
+      # Enclose the supplied command with the Powershell wrapper.
+      #
+      # @param command [String] optional repo target for built packages defined
+      #   at the project level
+      # @return [String] Wrapped Powershell Command
+      def powershell_command(command)
+        result = "powershell.exe -NoProfile -ExecutionPolicy Unrestricted -InputFormat None -Command #{command}"
+        result
+      end
+
+      # Given a host, path to MSI and MSI options, will create a batch file
+      #   on the host, returning the path to the randomized batch file and
+      #   the randomized log file
+      # @param [Host] host An object implementing {Beaker::Hosts}'s interface.
+      # @param [String] msi_path The path of the MSI - can be a local Windows style file path like
+      #                   c:\temp\puppet.msi OR a url like https://download.com/puppet.msi or file://c:\temp\puppet.msi
+      # @param  [Hash{String=>String}] msi_opts MSI installer options
+      #                   See https://docs.puppetlabs.com/guides/install_puppet/install_windows.html#msi-properties
+      # @api private
+      def install_chocolately_on_host(url)
+        timestamp = Time.new.strftime('%Y-%m-%d_%H.%M.%S')
+
+        batch_name = "install-chocolatey-#{timestamp}.ps1"
+        batch_path = "./#{batch_name}"
+
+        log_path = "\\install-puppet-#{timestamp}.log"
+
+        Tempfile.open(batch_name) do |tmp_file|
+          batch_contents = install_chocolately_ps1(url)
+
+          File.open(tmp_file.path, 'w') { |file| file.puts(batch_contents) }
+
+          require 'pry'
+          binding.pry
+
+          #Vanagon::Utilities.rsync_to("#{batch_path}", "#{@target_user}@#{@target}", "xxxxx", @ssh_port)
+          #Vanagon::Engine::Base.ship_workdir("./#{batch_path}")
+        end
+
+
+        return batch_path, log_path
```

Comment:
`install_chocolately_on_host` returns an array of two paths but it has a name that implies it's more of an imperative than a builder/constructor method. It also implicitly returns an array, but if you're explicitly calling `return` you should make sure that the object you return is also explicit:

```ruby
>> def some_thing(x, y)
|    return x, y
|  end  
=> :some_thing
>> some_thing true, false
=> [
    [0] true,
    [1] false
]
```

The name should probably better reflect what it's doing, and the return value should probably be explicit.